File: src/main/java/refactown/cleancode/c08boundary/java/LogonController.java
package refactown.cleancode.c08boundary.java;

public class LogonController {
	
	private final JsonParser jsonParser;

	public LogonController(JsonParser jsonParser) {
		super();
		this.jsonParser = jsonParser;
	}

	public User logon(String login, String password) {
		String userJson = callHttp(login, password);
		return (User) jsonParser.fromJson(userJson, User.class);
	}
	
    private String callHttp(String login, String password) {
        return "{\"login\":\"admin\", \"role\":\"admin\"}"; //MOCK
    }

}

File: src/main/kotlin/refactown/cleancode/c08boundary/LogonController.kt
package refactown.cleancode.c08boundary

class LogonController(private val jsonParser: JsonParser){

    fun logon(login: String, password: String): User{
        val userJson = callHttp(login, password)
        return jsonParser.fromJson(userJson, User::class.java) as User
    }

    private fun callHttp(login: String, password: String): String {
        return "{\"login\":\"admin\", \"role\":\"admin\"}" //MOCK
    }
}

data class User (val login: String, val role: String)
